tests/fill_poll.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+from fastapi.exceptions import ValidationException
+
+from core.server import server
+from models.models import User, Answer, QuestionTextbox, QuestionOption, FilledPoll
+
+FILE_PATH = "file.json"
+
+
+class TestFillPoll(unittest.TestCase):
+    def test_fill_poll(self):
+        poll = server.load_poll(FILE_PATH)
+        user1 = User(name="oilymacaroni")
+        user2 = User(name="among us")
+
+        server.game.register_user(user1)
+        server.game.register_user(user2)
+
+        server.game.start_polling_phase(poll)
+
+        answer1 = Answer(question_name="q1", value=[QuestionOption() for _ in range(2)])
+        answer2 = Answer(question_name="q2", value=QuestionTextbox())
+        filled_poll = FilledPoll(
+            answers=[answer1, answer2], user_about=user1, user_filling=user2
+        )
+
+        validation = server.game.validate_answers(filled_poll)
+        result = server.game.add_answer(filled_poll)
+        self.assertTrue(validation is None)
+        self.assertTrue(result)
+        self.assertTrue(server.game.user_data[user2][user1] is not None)
+        self.assertTrue(len(server.game.user_data[user2][user1]) == 2)
+
+    def test_fill_pol_bad_time(self):
```

Comment:
typo + test nie przechodzi, chyba brakuje game.start_polling_phase?
Nazwa testu też powinna wskazywać na to co on robi, a tu nie mam pojęcia

Reply:
już poprawiłem: zamiast wykrywać błąd nadesłania w złym czasie, po prostu nim rzucałem xD